### PR TITLE
Add Elasticsearch dashboard helper methods and constants

### PR DESCRIPTION
**Pull request by raihankhan**
Signed-off-by: raihankhan <raihan@appscode.com>

### CODE REVIEW

File: apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (ed ElasticsearchDashboard) ResourcePlural() string {
 }
 
 // CertificateName returns the default certificate name and/or certificate secret name for a certificate alias
-func (ed *ElasticsearchDashboard) CertificateName(alias ElasticsearchDashboardCertificateAlias) string {
+func (ed *ElasticsearchDashboard) CertificateName(alias ElasticsearchDashboardSecretAlias) string {
```
Comment (kamolhasan) [inline apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go:75]: CertificateName() --> CertificateSecretName()

Comment (kamolhasan) [reply apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go:75]: Keep `ElasticsearchDashboardCertificateAlias` as input params

Comment (kamolhasan) [reply apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go:75]: ElasticsearchDashboardConfigAlias = `operator`, `user`

```
GetConfigSecretName(alias) string{
 if alias== operator{
 	return  util.Namewithsuffix(ed.name, "config")
 } 
 
 if ed.spec.configsecret.name!=nil {
 return ed.spec.configsecret.name
} 
return  util.Namewithsuffix(ed.name, "user-config")
}
```

File: apis/dashboard/v1alpha1/types.go [hunk 2]
```diff
@@ -91,3 +77,14 @@ const (
 	StateYellow DashboardServerState = "yellow"
 	StateRed    DashboardServerState = "red"
 )
+
+// +kubebuilder:validation:Enum=dashboard-custom-config;dashboard-temp-config;dashboard-config;kibana-server;database-client
+type DashboardVolumeName string
+
+const (
+	CustomConfig            DashboardVolumeName = "dashboard-custom-config"
```
Comment (kamolhasan) [inline apis/dashboard/v1alpha1/types.go:85]: Start with prefix `DashboardVolume....`

**Review by kamolhasan — CHANGES_REQUESTED** [review]

File: apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go (hunk 1 above)
Comment (raihankhan) [reply apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go]: fixed

File: apis/dashboard/v1alpha1/types.go (hunk 2 above)
Comment (raihankhan) [reply apis/dashboard/v1alpha1/types.go]: fixed

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Pull request merged**